src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSpark.java

```diff
@@ -288,12 +289,15 @@ public int getPartition(Object key) {
 
     @Override
     protected void runTool(final JavaSparkContext ctx) {
-        // Check if we are using multiple inputs that the headers are all in the correct querygrouped ordering
+        final SAMFileHeader header = getHeaderForReads();
```

Comment:
This gets a merged header -- what will the underlying `SamFileHeaderMerger` do in the case where the sort order does not agree across headers? Will it throw in that case?

Reply:
Also, rename `header` to `mergedHeader`